source/system/native/MutexUnix.ooc

```diff
@@ -19,7 +19,17 @@ version(unix || apple) {
 			super()
 		}
 		lock: override func {
-			pthread_mutex_lock(this _backend&)
+			version(debugDeadlock) {
+				timeLimit_sec := 1
+				deadline: TimeSpec
+				currentTime: TimeVal
+				gettimeofday(currentTime&, null)
+				(deadline tv_sec, deadline tv_nsec) = (currentTime tv_sec + (timeLimit_sec as TimeT), (currentTime tv_usec * 1000) as Int)
```

Comment:
I don't understand. `tv_usec` is an `Int`, but you still cast it to `Int`.
You are storing it in `tv_nsec`, which is a `Long`. It would make more sense to cast to `Long`, but you don't have to since `Int->Long` is implicit.